src/adapter/enum_variant.rs

```diff
@@ -0,0 +1,304 @@
+use rustdoc_types::{Item, ItemEnum, Variant};
+use std::fmt;
+use std::num::ParseIntError;
+use std::sync::Arc;
+use std::{str::FromStr, sync::OnceLock};
+
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+pub(super) struct EnumVariant<'a> {
+    item: &'a Item,
+    discriminants: Arc<LazyDiscriminants<'a>>,
```

Comment:
We don't actually care about `Send` bounds, and we use `Rc` everywhere else in the adapter code. `cargo-semver-checks` runs queries to completion on the same thread only. 
```suggestion
    discriminants: Rc<LazyDiscriminants<'a>>,
```